spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -170,6 +170,20 @@ private final class Listener implements SchedulingAwareRunnable {
 
 		private final PulsarConsumerErrorHandler<T> pulsarConsumerErrorHandler;
 
+		private final boolean isBatchListener = this.containerProperties.isBatchListener();
+
+		private final boolean isManualAck = this.containerProperties.getAckMode().equals(AckMode.MANUAL);
```

Comment:
I think holding onto every possible combination of field from the CP that we need is going to grow and make this harder to maintain and less readable. Wdyt instead of simply holding onto the piece of data rather than the query on the data? In the case of the `AckMode`, we would not have 3 fields but rather 1 for the AckMode. Then the code would read how it did before with `this.ackMode...` vs. `this.containerProperties.getAckMode()...`.